internal/manifests/configmap/build.go

```diff
@@ -0,0 +1,22 @@
+package configmap
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// New returns a new k8s configmap
+func New(name, namespace string, labels map[string]string, data map[string]string) *corev1.ConfigMap {
```

Comment:
out of curiosity, why did you not put this in the `configmap.go` file?

Reply:
It all started to have separate file for the builder functions and a separate file for the k8s CRUD+CompareFunc/MutateFunc functions. Most packages have only simple `New()` factory functions as they don't require the builder pattern, because they are used by a single package, e.g. `internal/elasticsearch`. Others used by `internal/kibana` and `internal/elasticsearch` tend most of the times to require a builder. Thus, separating factories/builder in a file feels more hygiene.

Reply:
but now we have a ton of extra packages instead of containing it all to a single file with a focus

Reply:
The main entrypoint package for all k8s handling code is now in `internal/manifests` and that's a single package. It contains one subpackage (e.g. `configmap`) per k8s resource to make function naming shorter, i.e. instead of having them all being named as `CreateOrUpdateConfigMap`, `NewConfigMap`, `CompareConfigMapEqual` it is common practice to pull the common string `configmap` and group them in a separate pacakge. This looks and feels more coherent on the caller site:
- `configmap.New` instead of `manifests.NewConfigMap` or `k8shandler.NewConfigMap`
- `confitmap.CreateOrUpdate` instead of `manifests.CreateOrUpdateConfigmap` or `k8shandler.CreateOrUpdateConfigmap`
- `configmap.CompareEqual` instead of `manifests.CompareConfigMapEqual` or `k8shandler.CompareConfigMapEqual`